### PR TITLE
Update README.md formatting

### PR DESCRIPTION
**Pull request by alecbcs**
Update our README.md to include helpful links at the top, simplify badges to changing information, and center/increase logo size.

## New Preview Below

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,23 @@
-# <img src="https://cdn.rawgit.com/spack/spack/develop/share/spack/logo/spack-logo.svg" width="64" valign="middle" alt="Spack"/> Spack
-
-[![Unit Tests](https://github.com/spack/spack/workflows/linux%20tests/badge.svg)](https://github.com/spack/spack/actions)
-[![Bootstrapping](https://github.com/spack/spack/actions/workflows/bootstrap.yml/badge.svg)](https://github.com/spack/spack/actions/workflows/bootstrap.yml)
-[![codecov](https://codecov.io/gh/spack/spack/branch/develop/graph/badge.svg)](https://codecov.io/gh/spack/spack)
-[![Containers](https://github.com/spack/spack/actions/workflows/build-containers.yml/badge.svg)](https://github.com/spack/spack/actions/workflows/build-containers.yml)
-[![Read the Docs](https://readthedocs.org/projects/spack/badge/?version=latest)](https://spack.readthedocs.io)
```
Comment (tgamblin) [inline README.md:7]: some of these I'd rather not remove b/c they're important for us to keep track of -- specifically bootstrapping and RTD.

File: README.md [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
-[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
-[![Slack](https://slack.spack.io/badge.svg)](https://slack.spack.io)
-[![Matrix](https://img.shields.io/matrix/spack-space%3Amatrix.org?label=Matrix)](https://matrix.to/#/#spack-space:matrix.org)
+<div align="center">
```
Comment (tgamblin) [inline README.md]: I like the bigger logo but I think centering is going to look bad with more badges (which I at least want to keep).  Can we align left?

File: README.md [hunk 3]
```diff
@@ -1,13 +1,23 @@
...
+<img src="https://cdn.rawgit.com/spack/spack/develop/share/spack/logo/spack-logo-text.svg" width="250" valign="middle" alt="Spack"/>
+<br clear="all">
+<a href="https://github.com/spack/spack/actions/workflows/ci.yml"><img src="https://github.com/spack/spack/workflows/ci/badge.svg" alt="CI Status"></a>
+<a href="https://github.com/spack/spack/actions/workflows/build-containers.yml"><img src="https://github.com/spack/spack/actions/workflows/build-containers.yml/badge.svg" alt="Containers Status"></a>
+<a href="https://codecov.io/gh/spack/spack"><img src="https://codecov.io/gh/spack/spack/branch/develop/graph/badge.svg" alt="Code coverage"/></a>
+<a href="https://slack.spack.io"><img src="https://slack.spack.io/badge.svg" alt="Slack"/></a>
+<a href="https://matrix.to/#/#spack-space:matrix.org"><img src="https://img.shields.io/matrix/spack-space%3Amatrix.org?label=matrix" alt="Matrix"/></a>
+</h1>
+
+**[Getting Started] &nbsp; • &nbsp; [Config] &nbsp; • &nbsp; [Community] &nbsp; • &nbsp; [Contributing] &nbsp; • &nbsp; [Packaging Guide]**
```
Comment (tgamblin) [inline README.md:15]: Making the packaging guide more prominent is great - should probably be added to the contributing section as well.  Is there a way to make this look nice without the &nbsp's?

**Review by tgamblin — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (alecbcs) [reply README.md:15]: Without the `&nbsp;`'s html will just concatenate the spaces. There may be a way to put in a hidden/invisible unicode character if you'd prefer that. Although I thought that might lead to a confusing edit if someone doesn't know they're in the line for formatting and removes one unintentionally.

File: README.md (hunk 2 above)
Comment (alecbcs) [reply README.md]: For sure, I'll change the alignment back to be left.

File: README.md (hunk 1 above)
Comment (alecbcs) [reply README.md:7]: For sure. Added back in.

**Review by tgamblin — APPROVED** [review]

**Pull request merged**